### PR TITLE
bug fix for RealtimePublisher with NON_POLLING

### PR DESCRIPTION
**Pull request by YoavFekete**
its a fix bug for the issues reported in #43  and #82 
added a new unit tests that test the Non)Polling version

### CODE REVIEW

File: include/realtime_tools/realtime_publisher.h [hunk 1]
```diff
@@ -165,14 +165,21 @@ class RealtimePublisher
     is_running_ = true;
     turn_ = REALTIME;
 
+
```
Comment (destogl) [inline include/realtime_tools/realtime_publisher.h:168]:
```suggestion
```

File: test/realtime_publisher_tests_non_polling.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019, Open Source Robotics Foundation, Inc.
```
Comment (destogl) [inline test/realtime_publisher_tests_non_polling.cpp:2]: Copyright should be updated here. for example: "ros2_control development team" can be used as generic copyright holder.

**Review by destogl — APPROVED** [review]
This looks good to me. When do we want to use this?

File: test/realtime_publisher_tests_non_polling.cpp (hunk 2 above)
Comment (YoavFekete) [reply test/realtime_publisher_tests_non_polling.cpp]: done

**Comment by YoavFekete** [comment]
@bmagyar   any idea when and if this will be merged?
i want to continue to work on the publisher refactoring, but i gues bteer first finish with this

**Comment by YoavFekete** [comment]
@destogl ok i think fixed all  the workflow run errors . can u please rerun the tests

**Comment by YoavFekete** [comment]
> @destogl ok i think fixed all the workflow run errors . can u please rerun the tests

ok lets try again

**Review by bmagyar — APPROVED** [review]

**Pull request merged**